In [343]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.preprocessing import MinMaxScaler,PolynomialFeatures
import seaborn as sns 
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
from sklearn.preprocessing import LabelEncoder


In [344]:
df=pd.read_csv("cleaned.csv")


In [345]:
correlation_matrix = df.corr()


In [346]:
#ast column is the target variable
target_variable_name = df.columns[-1]

# Get the correlation of each feature with the target variable
correlation_with_target = correlation_matrix[target_variable_name].drop(target_variable_name)

# Sort correlations by absolute value
sorted_correlation_with_target = correlation_with_target.abs().sort_values(ascending=False)


In [347]:

# Select the top 10 most correlated features
top_features = sorted_correlation_with_target.head(30).index
new_df=df[top_features]


In [348]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 30 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   OverallQual   1460 non-null   int64
 1   GrLivArea     1460 non-null   int64
 2   GarageCars    1460 non-null   int64
 3   GarageArea    1460 non-null   int64
 4   ExterQual     1460 non-null   int64
 5   BsmtQual      1460 non-null   int64
 6   TotalBsmtSF   1460 non-null   int64
 7   1stFlrSF      1460 non-null   int64
 8   KitchenQual   1460 non-null   int64
 9   FullBath      1460 non-null   int64
 10  GarageFinish  1460 non-null   int64
 11  YearBuilt     1460 non-null   int64
 12  TotRmsAbvGrd  1460 non-null   int64
 13  YearRemodAdd  1460 non-null   int64
 14  Fireplaces    1460 non-null   int64
 15  MasVnrArea    1460 non-null   int64
 16  FireplaceQu   1460 non-null   int64
 17  LotArea       1460 non-null   int64
 18  GarageType    1460 non-null   int64
 19  HeatingQC     1460 non-null

In [349]:
Y=df["SalePrice"]

In [350]:
Y=pd.DataFrame(Y)

In [351]:
model=LinearRegression()

In [352]:
scaler=MinMaxScaler()

In [353]:
poly=PolynomialFeatures(degree=1)

In [354]:
new_df=poly.fit_transform(new_df)

In [355]:
new_df.shape

(1460, 31)

In [356]:
new_df_scaled=scaler.fit_transform(new_df)

In [357]:
Y_scaled=scaler.fit_transform(Y)

In [358]:
model.fit(new_df_scaled,Y_scaled)

LinearRegression()

In [359]:
Y_pre=model.predict(new_df_scaled)

In [360]:
Y_pre = scaler.inverse_transform(Y_pre)

In [361]:
mse=mean_squared_error(Y,Y_pre)/2

In [362]:
r2=r2_score(Y,Y_pre)

In [363]:
print("\nMean Squared Error (MSE):", mse)
print("R-squared (R2) Score:", r2)



Mean Squared Error (MSE): 526563302.66370517
R-squared (R2) Score: 0.8330169798661353



##

with 1 degree in polynomial regression



Mean Squared Error (MSE): 526563302.66370517


R-squared (R2) Score: 0.8330169798661353

##

##
with 2 degree in polynomial regression

Mean Squared Error (MSE): 185454416.22640982

R-squared (R2) Score: 0.941188954182729


##

##
with 3 degree in polynomial regression

Mean Squared Error (MSE): 9631.84931582204

R-squared (R2) Score: 0.9999969455613786
##
##


with 4 degree in polynomial regression

Mean Squared Error (MSE): 9631.849340913714


R-squared (R2) Score: 0.9999969455613708

##



